docs/DeveloperGuide.md

```diff
@@ -504,3 +466,54 @@ testers are expected to do more *exploratory* testing.
+Most of the difficulty in our project stems from the challenge of trying to **integrate mutable objects into an established immutable project**. While we knew the project was developed based on the principles of immutability, we felt in our case, it was better to use mutable objects for insurance plans and claims as the lists and claims were objects that were changed frequently unlike details that were currently found in addressbook level-3 such as names, emails and phone numbers. This would allow future developers to also work more easily with the given code base as adding and removing plans, introducing new insurance related features would be much more easily implemented with less code. We also wanted to challenge ourselves to see if this was possible at all.
+
+The **biggest hurdle** to overcome with this project was **modifying the testing methods**. The test methods were mostly designed for immutable objects and there were strict tests like JsonSerializableAddressBookTest which were the most difficult to pass. This is because the test compares the hardcoded file of clients to a generated file of clients. However, at first since we were making the objects mutable, the generated file becomes different in comparison to the hardcoded file ONLY during the full testing sequence. When the test was done individually, the test passes. We scratched our heads over this for a very long time and eventually managed to solve this problem by changing the equals method for `Client` object and also the `InsurancePlansManager` object.
+
+However, **this fix initially caused other tests break**. This includes tests like `addClaimCommandTest.java` as the `assertCommandSuccess` method in `TestUtil.java` did not compare actual and expected models correctly. This meant that even if there were some changes in the insurance plan manager, sometimes the two models are evaluated as equal to each other. We then had to create new assert methods for commands (like assertInsuranceCommandSuccess) for all the insurance plan related objects.
+
+Our implementations were at **least as difficult compared to half of AB-3 effort**. While we did not have to create many huge structures from scratch like the `ModelManager`, `Logic` and `Storage` components, we had to work with moving parts compared to AB-3. AB-3 does not support auto-sort of clients. However, we wanted a quick and easy way for insurance managers to view their most critical claims and as such auto sorted claims for each client to show open claims first.
+
+Lastly, as **our project involved money**, which is a sensitive value that we must get right. We decided to store all values in cents to avoid having to use float (and having issues with floating point accuracy). However, this led to **greater care required to parse and convert values** to cents and ended up being a very time consuming method to unit test. This method was still not completely fixed (the constraint of 1million was yet to be implemented).
```

Comment:
What if we phrase it as `... sensitive value that has to be exact` instead of `...sensitive value that we must get right`?

Reply:
changed to `sensitive value that has to be precise and accurate`

Reply:
Can!